cmd/snap-confine/snap-confine.c

```diff
@@ -678,9 +679,10 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 	}
 	if (!sc_cgroup_is_v2()) {
 		sc_cgroup_freezer_join(inv->snap_instance, getpid());
-		if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
-			sc_cgroup_pids_join(inv->security_tag, getpid());
-		}
+	}
+	// TODO: This should be done uniformly for classic and non-classic snaps.
+	if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
+		sc_cgroup_snapd_hierarchy_join(inv->security_tag, getpid());
 	}
```

Comment:
I agree with @bboozzoo that we should just unconditionally join now, in this PR. Note that the logic of this isn't quite right on a pure v2 system due to my previous comment.

Reply:
The logic isn't wrong as was explained above (in pure v2 it works just the same).

For the purpose of respecting the mechanism of feature flags I would like _not_ to join unconditionally in this pull request and perform this change separately.